### PR TITLE
doc/cephfs: line-edit "Mirroring Module"

### PR DESCRIPTION
**Pull request by zdover23**
Line-edit the "Mirroring Module" section of
doc/cephfs/cephfs-mirroring.rst. Add prompts and formatting where such things contribute to the realization of adequate sentences.

This commit is a follow-up to https://github.com/ceph/ceph/pull/51505.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/cephfs/cephfs-mirroring.rst [hunk 1]
```diff
@@ -42,129 +44,175 @@ The mirror daemon should be spawned using `systemctl(1)` unit files::
...
   $ cephfs-mirror --id mirror --cluster site-a -f
 
-.. note:: The user specified here is `mirror` created in the `Creating Users` section.
+.. note:: The user specified here is `mirror`, the creation of which is
+   described in the :ref:`Creating Users<cephfs_mirroring_creating_users>`
+   section.
+
+Multiple ``cephfs-mirror`` daemons may be deployed for concurrent
+synchronization and high availability. Mirror daemons share the synchronization
+load using a simple ``M/N`` policy, where ``M`` is the number of directories
```
Comment (anthonyeleven) [inline doc/cephfs/cephfs-mirroring.rst:53]: I might say "workload" but this isn't wrong.

File: doc/cephfs/cephfs-mirroring.rst [hunk 2]
```diff
@@ -42,129 +44,175 @@ The mirror daemon should be spawned using `systemctl(1)` unit files::
...
+.. prompt:: bash $
 
-  $ ceph fs snapshot mirror peer_add primary_fs client.mirror_remote@remote_ceph remote_fs
+   ceph fs snapshot mirror peer_add primary_fs client.mirror_remote@remote_ceph remote_fs
 
-To avoid maintaining remote cluster configuration file and remote ceph user keyring in the primary
-cluster, users can bootstrap a peer (which stores the relevant remote cluster details in the monitor
-config store on the parimary cluster). See `Bootstrap Peers` section.
+To avoid having to maintain the remote cluster configuration file and remote
+ceph user keyring in the primary cluster, users can bootstrap a peer (which
```
Comment (anthonyeleven) [inline doc/cephfs/cephfs-mirroring.rst:154]: Ceph

Users, or admins?

File: doc/cephfs/cephfs-mirroring.rst [hunk 3]
```diff
@@ -42,129 +44,175 @@ The mirror daemon should be spawned using `systemctl(1)` unit files::
...
+cluster).
 
-`<remote_cluster_spec>` is of format `client.<id>@<cluster_name>`.
-`<remote_fs_name>` is optional, and defaults to `<fs_name>` (on the remote cluster).
+For this command to succeed, the remote cluster's Ceph configuration and user
+keyring must be available in the primary cluster. For example, if a user named
+``client_mirror`` is created on the remote cluster which has ``rwps``
+permissions for the remote file system named ``remote_fs`` (see `Creating
+Users`) and the remote cluster is named ``remote_ceph`` (that is, the remote
+cluster configuration file is named ``remote_ceph.conf`` on the primary
```
Comment (anthonyeleven) [inline doc/cephfs/cephfs-mirroring.rst:145]: Ambiguity here caused people in the past to think they needed custom cluster names with rbd-mirror.  Nice to see this made explicit.

File: doc/cephfs/cephfs-mirroring.rst [hunk 4]
```diff
@@ -42,129 +44,175 @@ The mirror daemon should be spawned using `systemctl(1)` unit files::
...
+   ceph fs snapshot mirror add <fs_name> <path>
 
-e.g.::
+To stop mirroring directory snapshots, run a command of the following form:
+
+.. prompt:: bash $
+
+   ceph fs snapshot mirror remove <fs_name> <path>
+
+Only absolute directory paths are allowed.
```
Comment (anthonyeleven) [inline doc/cephfs/cephfs-mirroring.rst:189]: No .. for you

File: doc/cephfs/cephfs-mirroring.rst [hunk 5]
```diff
@@ -42,129 +44,175 @@ The mirror daemon should be spawned using `systemctl(1)` unit files::
...
+To stop mirroring directory snapshots, run a command of the following form:
+
+.. prompt:: bash $
+
+   ceph fs snapshot mirror remove <fs_name> <path>
+
+Only absolute directory paths are allowed. 
+
+Paths are normalized by the mirroring module. This means that ``/a/b/../b`` is
+equivalent to ``/a/b``. Paths always start from the cephfs file-system root and
```
Comment (anthonyeleven) [inline doc/cephfs/cephfs-mirroring.rst:192]: CephFS?

**Review by anthonyeleven — APPROVED** [review]
Non-blocking suggestions

File: doc/cephfs/cephfs-mirroring.rst (hunk 5 above)
Comment (zdover23) [reply doc/cephfs/cephfs-mirroring.rst]: Yeah. Good one.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/51542 - Reef backport
https://github.com/ceph/ceph/pull/51543 - Quincy backport
https://github.com/ceph/ceph/pull/51544 - Pacific backport

**Pull request merged**